app/services/parse/google_parse_service.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Parse
+  class GoogleParseService
+    SELECTORS = {
+      adwords_top: 'div.pla-unit-container a.pla-unit-title-link',
+      adwords_body: 'div.data-text-ad div.v5yQqb',
+      non_adwords: 'div#search div.yuRUbf a[data-ved]',
+      href: 'href'
+    }.freeze
+
+    def initialize(html_code)
+      @html_code = html_code
+      raise IcRailsHueyToby::Errors::ParseServiceError unless html_code
+    end
+
+    def parse
+      parse_html
+    end
```

Comment:
Optional: Remove this, and rename `parse_html` to `parse` and make the function public